lib/ecto/adapters/sql.ex

```diff
@@ -570,33 +538,6 @@ defmodule Ecto.Adapters.SQL do
     end
   end
 
-  defp process_row(row, process, fields, sources) do
```

Comment:
The difference in this code and the previous code is that adapters are no longer responsible for the whole logic below. To handle this in a backwards compatible fashion, you can change this code in v2.1:

```
mapper = &process_row(&1, process, fields, sources)
```

to something like this:

```
mapper = processor(process, fields, sources)
```

where

```
defp processor(process, fields, source) when is_function(process, 3) do
  &process_row(&1, process, fields, sources)
end
defp processor(process, _fields, _source) when is_function(process, 1) do
  process
end
defp processor(nil, _fields, _source) do
  nil
end
```